### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by beatngu13**
Please go commit by commit.

### CODE REVIEW

File: src/main/java/de/retest/recheck/junit/jupiter/RecheckExtension.java [hunk 1]
```diff
@@ -79,20 +80,18 @@ private void execute( final Consumer<RecheckLifecycle> lifecycleMethod, final Ob
...
-			lock( field, accessibility );
-		}
+		return applyOnFieldValue( field, testInstance, value -> value instanceof RecheckLifecycle );
 	}
 
 	private RecheckLifecycle getRecheckLifecycle( final Field field, final Object testInstance ) {
+		return applyOnFieldValue( field, testInstance, value -> (RecheckLifecycle) value );
+	}
+
+	private <T> T applyOnFieldValue( final Field field, final Object testInstance, final Function<Object, T> f ) {
```
Comment (briemla) [inline src/main/java/de/retest/recheck/junit/jupiter/RecheckExtension.java]: Nice one. Didn't think about that when implementing the extension. Could we find a better name for `f`?

Comment (beatngu13) [reply src/main/java/de/retest/recheck/junit/jupiter/RecheckExtension.java]: `fun` (like in Kotlin), `func`, `function`, `toApply`, … ?

Comment (briemla) [reply src/main/java/de/retest/recheck/junit/jupiter/RecheckExtension.java]: What about `converter`?
Otherwise I would vote for `toApply`.

Comment (beatngu13) [reply src/main/java/de/retest/recheck/junit/jupiter/RecheckExtension.java]: If I rename the method to `mapFieldValue` and then name the function param `mapper`? But we can also ge for `convertFieldValue` and `converter`.

Comment (briemla) [reply src/main/java/de/retest/recheck/junit/jupiter/RecheckExtension.java]: I prefer the `map` variant, as it is more common in processing `Stream`s and `Optional`s.

**Pull request merged**